cgra/CGRAKingMeshRTL.py

```diff
@@ -149,11 +149,23 @@ def construct( s, DataType, PredicateType, CtrlType, width, height,
         s.tile[i].to_mem_waddr.rdy //= 0
         s.tile[i].to_mem_wdata.rdy //= 0
 
-  # Line trace
-  def line_trace( s ):
-    # str = "||".join([ x.element.line_trace() for x in s.tile ])
-    # str += " :: [" + s.data_mem.line_trace() + "]"
-    res = "||\n".join([ (("[tile"+str(i)+"]: ") + x.line_trace() + x.ctrl_mem.line_trace())
-                              for (i,x) in enumerate(s.tile) ])
-    res += "\n :: [" + s.data_mem.line_trace() + "]    \n"
+  # verbose trace if verbosity > 0
+  def verbose_trace(s, verbosity=1):
+    res = ''
+    for (i, x) in enumerate(s.tile):
+      res += "# [tile" + str(i) + "]: " + x.line_trace(verbosity=verbosity) + x.ctrl_mem.line_trace(verbosity=verbosity) + '\n'

```

Comment:
I didn't get this part. It looks if `verbose_trace(1)`, this line would call `line_trace(1)`, then `line_trace(1)` would call `verbose_trace(1)`?

Reply:
Yes, not all classes have `verbose_trace` now, if there is, `line_trace `will call `verbose_trace`, or will run `line_trace` itself.

Reply:
Can the `verbose_trace()` just have:
```
res += "# [tile" + str(i) + "]: " + x.verbose_trace(verbosity=verbosity) + x.ctrl_mem.verbose_trace(verbosity=verbosity) + '\n'
```
instead of 
```
res += "# [tile" + str(i) + "]: " + x.line_trace(verbosity=verbosity) + x.ctrl_mem.line_trace(verbosity=verbosity) + '\n'
```
which is more intuitive?
For the modules/classes that haven't designed `verbose_trace()`, you can have:
```
def verbose_trace(s, verbosity = 1):
  s.line_trace(verbosity)
```
WDYT?

Reply:
Or it's better to add verbose trace to every class?

Reply:
How about now, add a trace entrance, but in *_test.py, function name has to be line_trace().

Reply:
done